plugins/modules/zos_tso_command.py

```diff
@@ -128,9 +146,14 @@ def copy_rexx_and_run_commands(script, commands, module):
         rc, stdout, stderr = module.run_command([tmp_file.name, command])
         command_results = {}
         command_results["command"] = command
+        command_results["origrc"] = rc
+        if rc <= max_rc:
+            rc = 0
```

Comment:
This looks a bit off here, suppose max_rc = 16 and a rc 8 is returned , thus its a success but in line 151 the RC is set to 0 instead of 8, is how this reads to me.

Reply:
This is intentional.  origrc is for the original return code from this command.  If rc is non-0 it could can break upstream stuff.  So I've switched it to 0 if the rc is below max_rc, but added origrc to preserve the actual return code.
In your example, origrc would be 8, and rc set to 0, since the return code was below max_rc.

Reply:
To clarify, I've renamed the value to orig_rc, and added a brief description in the return value doc.

Reply:
Can we have a discussion about this, feel the RC returned should be what comes back no matter its less than or greater than max_rc.

For `zos_job_submit` what we did was return the max_rc the user provided and then returned the RC, we had logic to compare max_rc to rc , on success we do this:
```
    result["changed"] = True if is_changed else False
    result["failed"] = False
    module.exit_json(**result)
```
On failure we would have it:
```
   if job_rc > max_rc:
        raise Exception("The job return code, 'ret_code[code]' {0} for the submitted job is "
                        "greater than the value set for option 'max_rc' {1}. "
                        "Increase the value for 'max_rc' otherwise this job submission "
                        "has failed.".format(str(job_rc), str(max_rc)))
```
which then falls down into our final catch and that finishes building up the response:
```
    except Exception as err:
        result["failed"] = True
        result["changed"] = False
        result["msg"] = ("The JCL submitted with job id {0} but "
                         "there was an error, please review "
                         "the error for further details: {1}".format
                         (str(job_submitted_id), str(err)))
        module.exit_json(**result)
```

The code between the two modules don't have to look the same, but I think the behaviors should be the same and unsure why it would cause issues 'upstream' , we choose when to exit and tell ansible `result["failed"] = True`